linera-core/src/chain_worker/state.rs

```diff
@@ -1071,20 +1109,34 @@ where
                 .await;
         }
 
-        let blobs_in_block = self.state.get_blobs(block.blob_ids()).await?;
+        let published_blobs_in_block = self.state.get_blobs(block.published_blob_ids()).await?;
```

Comment:
We need to write _all_ blobs that we don't have yet, published or read.

Reply:
So for the reads, if we don't have the blob that we're trying to read, it would have already failed the block at this point, no? So then the blobs we're reading should already be in storage at this point.
So the only thing we need to actually store is the blobs we're publishing. Or what am I missing? 🤔

Reply:
Just like with the published blobs, the ones we're missing could be among the `hashed_blobs` provided by the client.

As discussed, in `process_confirmed_value` there shouldn't be a difference between published and read blobs.